frontend/src/modules/Metadata_Forms/components/fields.js

```diff
@@ -0,0 +1,129 @@
+import {
+  Autocomplete,
+  Checkbox,
+  FormControlLabel,
+  TextField
+} from '@mui/material';
+import PropTypes from 'prop-types';
+import React, { useEffect, useState } from 'react';
+import { getGlossaryTree } from '../../Glossaries/services';
+import { SET_ERROR } from '../../../globalErrors';
+
+export const FreeInputField = (props) => {
+  const { field, onChange, errors } = props;
+  return (
+    <TextField
+      sx={{ width: '100%' }}
+      label={field.name}
+      error={errors[field.name]}
+      name={field.name}
+      onKeyUp={(event) => {
+        onChange(event.target.value);
+      }}
+    ></TextField>
+  );
+};
+FreeInputField.propTypes = {
+  field: PropTypes.any.isRequired
+};
+
+export const BooleanField = (props) => {
+  const { field, onChange } = props;
+  return (
+    <FormControlLabel
+      sx={{ pl: 1 }}
+      control={
+        <Checkbox
+          id={field.name}
+          onChange={(event, checked) => onChange(checked)}
+        />
+      }
+      label={field.name}
+    />
+  );
+};
+
+BooleanField.propTypes = {
+  field: PropTypes.any.isRequired
+};
+
+export const GlossaryTermField = (props) => {
+  const { field, client, dispatch, onChange, errors } = props;
+
+  const [glossaryOptions, setGlossaryOptions] = useState([]);
+
+  const fetchGlossaryTerms = async () => {
+    const response = await client.query(
+      getGlossaryTree({
+        nodeUri: field.glossaryNodeUri,
+        filter: { pageSize: 500 }
```

Comment:
nit: maybe use `Defaults.selectListFilter` here

Reply:
Also should be able to pass `nodeType` variable to filter on `Term` rather than filtering the response (looking at `GlossaryNodeSearchFilter`)